[pytorch] add operator name to callBoxed() error message

ljk53 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#39562 [pytorch] add operator name to callBoxed() error message**

Improve error message to help debug missing boxed kernel error.

Differential Revision: [D21900540](https://our.internmc.facebook.com/intern/diff/D21900540/)

ljk53 committed:
[pytorch] add operator name to callBoxed() error message

Improve error message to help debug missing boxed kernel error.

Differential Revision: [D21900540](https://our.internmc.facebook.com/intern/diff/D21900540/)

[ghstack-poisoned]

ljk53 commented on aten/src/ATen/core/boxing/KernelFunction_impl.h:
```diff
@@ -34,10 +34,16 @@ inline bool KernelFunction::isFallthrough() const {
+            TORCH_INTERNAL_ASSERT(
+                false,
+                "Tried to call KernelFunction::callBoxed() on an uninitialized KernelFunction. ",
+                dumpOperatorHandle(opHandle));
```
OperatorHandle is only forward declared here - Dispatcher.h includes this header, so creating a dump method to break the circular dep.

ezyang replied:
Here's an alternative which is better for binary size: move the error reporting functionality into the cpp file (so lines 36 to 48 as seen here. Then you don't have to deal with the forward declaration, and we don't repeatedly inline the error reporting code here.

ljk53 replied:
Updated - also added some mobile selective build specific info as this is where it frequently happens - hope you don't feel it's too verbose. cc: @smessmer

ezyang reviewed (approved):
I don't think it's worth bikeshedding too much, but I would definitely like it if we did the refactor I suggested.

ljk53 committed:
Update on "[pytorch] add operator name to callBoxed() error message"


Improve error message to help debug missing boxed kernel error.

Differential Revision: [D21900540](https://our.internmc.facebook.com/intern/diff/D21900540/)

[ghstack-poisoned]

smessmer commented on aten/src/ATen/core/boxing/KernelFunction.cpp:
```diff
@@ -49,4 +49,25 @@ bool KernelFunction::_equalsBoxedAndUnboxed(const KernelFunction& other) const {
+              "Tried to call KernelFunction::callBoxed() on an uninitialized KernelFunction.",
+              " opname: ",
+              opHandle.operator_name(),
+              " For mobile selective build please make sure to include all ops exported from `torch.jit.export_opnames(model)`.");
```
If somebody who doesn't know what a selective build is sees this, it might be confusing. Maybe say "If you're using mobile selective build" instead of "For mobile selective build" to make it more clear.

ljk53 committed:
Update on "[pytorch] add operator name to callBoxed() error message"


Improve error message to help debug missing boxed kernel error.

Differential Revision: [D21900540](https://our.internmc.facebook.com/intern/diff/D21900540/)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] add operator name to callBoxed() error message"


Improve error message to help debug missing boxed kernel error.

Differential Revision: [D21900540](https://our.internmc.facebook.com/intern/diff/D21900540/)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] add operator name to callBoxed() error message"


Improve error message to help debug missing boxed kernel error.

Differential Revision: [D21900540](https://our.internmc.facebook.com/intern/diff/D21900540/)

[ghstack-poisoned]
